engine/profiler/src/profile_render.cpp

```diff
@@ -641,35 +645,31 @@ namespace dmProfileRender
         if (!current_frame)
             return;
 
-        ProfilerThread* last_thread = GetSelectedThread(render_profile, render_profile->m_CurrentFrame);
-
         render_profile->m_CurrentFrame = DuplicateProfilerFrame(current_frame);
 
         ProfilerThread* current_thread = GetSelectedThread(render_profile, render_profile->m_CurrentFrame);
-        if (last_thread->m_NameHash != current_thread->m_NameHash)
+        if (last_thread_name != current_thread->m_NameHash)
         {
-            last_thread = 0;
             render_profile->m_MaxFrameTime = 0;
         }
 
-        //uint64_t last_frame_time = last_thread ? last_thread->m_SamplesTotalTime : 0;
         uint64_t this_frame_time = current_thread->m_SamplesTotalTime;
 
         bool new_peak_frame = render_profile->m_MaxFrameTime < this_frame_time;
         render_profile->m_MaxFrameTime = dmMath::Max(render_profile->m_MaxFrameTime, this_frame_time);
 
-        if (render_profile->m_Mode == PROFILER_MODE_SHOW_PEAK_FRAME)
+        if (new_peak_frame)
         {
-            if (new_peak_frame)
+            if (render_profile->m_LastPeakFrame)
             {
-                ProfilerFrame* snapshot = DuplicateProfilerFrame(render_profile->m_CurrentFrame);
-
-                FlushRecording(render_profile, 1);
-                render_profile->m_RecordBuffer.SetSize(1);
-                render_profile->m_RecordBuffer[0] = snapshot;
+                DeleteProfilerFrame(render_profile->m_LastPeakFrame);
             }
+            render_profile->m_LastPeakFrame = DuplicateProfilerFrame(render_profile->m_CurrentFrame);
```

Comment:
Add storing peak frame as separate struct field because:
1. it solves problem with display peak frame in case if it appeared before switching to `PROFILER_MODE_SHOW_PEAK_FRAME`
2. it solves problem when peak mode can conflict with record mode because they used the same record buffer. Now it can be safetly switched between recorded frames and peak frame.